ouroboros-network/src/Ouroboros/Network/Mux/Interface.hs

```diff
@@ -194,7 +194,7 @@ newtype Connection m = Connection {
 -- |
 -- CPS style for runing mux layer
 --
-type WithConnection m addr a r = addr -> (a -> m r) -> m r
+type WithConnection m addr conn r = addr -> (conn -> m r) -> m r
```

Comment:
I realise we're just changing names here, but I have to say I still don't get the purpose of this type alias or what it buys us. It's all rather confusing to use I think.

What does it buy us? What if we expanded out all these type aliases and newtypes, instead of:
```
      connect :: WithConnection m addr (Connection m) r
```
we'd have
```
      connect :: addr -> (m () -> m r) -> m r
```
And also what's the point in the `r` type param? Couldn't connect be locally polymorphic?
```
      -- | Given the address to connect to, the continuation is passed the action for executing the
      -- mux application for this connection. This allows for example bracketing with before/after
      -- actions, or one can simply use 'id'.
      --
      -- > connect peerAddr (\run -> bracket before after run)
      --
      connect :: forall r. addr -> (m () -> m r) -> m r
```

Reply:
I agree; I already removed `r` but not yet pushed.  @dcoutts do you prefer me to push on top of this PR the other changes I have or merge this one first?